sdk/identity/identity/src/msal/nodeFlows/msalClient.ts

```diff
@@ -184,12 +195,17 @@ To work with multiple accounts for the same Client ID and Tenant ID, please prov
       state.cachedAccount = accounts[0];
     }
 
+    // Keep track and reuse the claims we received across challenges
+    if (options.claims) {
+      state.cachedClaims = options.claims;
```

Comment:
do we have to check if these are the same / merge them? or can we always safely update?

Reply:
Good Q, the existing code always safely updates (or at least once I simplified it)

Old code: https://github.com/Azure/azure-sdk-for-js/blob/727a7208251961b5036d8e1d86edaa944c42e3d6/sdk/identity/identity/src/msal/nodeFlows/msalNodeCommon.ts#L440-L449

|options.claims|cachedClaims|request claims|
|----|----|-----|
|undefined|undefined|undefined|
|undefined|defined|cachedClaims|
|defined|undefined|options.claims|
|defined|defined|options.claims|


So this reads to me as `options.claims` always takes precedence, cached claims will be used if they exist and options.claims is undefined. But the old code is doing some unnecessary work. Let me know if you interpret it differently / check my work!